python-package/xgboost/spark/core.py

```diff
@@ -1148,7 +1151,7 @@ def _train_booster(
                 if dvalid is not None:
                     dval = [(dtrain, "training"), (dvalid, "validation")]
                 else:
-                    dval = None
+                    dval = [(dtrain, "training")]
```

Comment:
@trivialfis, Could you check this is ok by enabling it by default?

Reply:
Hi @trivialfis, I wonder if this default eval dataset is necessary?

Reply:
Will look into this.

Reply:
Looks good to me, it's unlikely someone will train a model without any evaluation.